app.js

```diff
@@ -379,161 +765,75 @@ function processAdditionalDataResponse(additionalDataResponse) {
-      this.geometry.coordinates[1] = [e.latlng.lng, e.latlng.lat];
-      var features = turf.featureCollection([
-        turf.point(this.geometry.coordinates[0]),
-        turf.point(this.geometry.coordinates[1])
-      ]);
-      var geoJsonData = turf.envelope(features);
-      this.redraw(geoJsonData);
-    },
-    function(event) {
-      this.geometry = {
-        type: "MultiPoint",
-        shapeType: "Rectangle",
-        coordinates: [[event.latlng.lng, event.latlng.lat]]
-      };
-      this.setOneClickMapListeners();
-    }
-  );
-});
+    modalContent.innerText = message;
+  modal.style.display = "block";
+  document.querySelector("body").addEventListener("click", function() {
+    modal.style.display = "none";
+    modal.classList.remove("error");
+  })
+}
 
-document.getElementById("corridor-button").addEventListener("click", function() {
-  drawHandler(
-    "corridor-form",
-    function(e) {
-      this.geometry.coordinates[this.geometry.coordinates.length - 1] = [
-        e.latlng.lng,
-        e.latlng.lat
-      ];
-      var geoJsonData = turf.buffer(
-        this.geometry,
-        this.geometry.radius,
-        turfOptions
-      );
-      this.redraw(geoJsonData);
-    },
-    function(event) {
-      var self = this;
-      this.geometry = {
-        type: "LineString",
-        shapeType: "Corridor",
-        radius: parseFloat(document.getElementById("corridor-radius").value),
-        coordinates: [
-          [event.latlng.lng, event.latlng.lat],
-          [event.latlng.lng, event.latlng.lat]
-        ]
-      };
-      this.setDblClickMapListeners();
-      map.on("dblclick", function() {
-        self.geometry.coordinates.pop();
-        self.endDrawing(self.geometry, self.geoJson);
-      });
-    }
-  );
-});
+function displayAdminKey(generatedAdminKey) {
+  document.getElementById("generated-admin-key").innerText = generatedAdminKey;
+  document.querySelector("label[for='secret']").style.display = "none";
+  document.querySelector("label[for='generated-admin-key']").style.display = "block";
+}
 
-document.getElementById("polygon-button").addEventListener("click", function() {
-  drawHandler(
-    null,
-    function(e) {
-      this.geometry.coordinates[this.geometry.coordinates.length - 1] = [
-        e.latlng.lng,
-        e.latlng.lat
-      ];
-      this.geometry.type = "LineString";
-      this.redraw(this.geometry);
-    },
-    function(event) {
-      var self = this;
-      this.geometry = {
-        coordinates: [
-          [event.latlng.lng, event.latlng.lat],
-          [event.latlng.lng, event.latlng.lat]
-        ]
-      };
-      this.setDblClickMapListeners();
-      map.on("dblclick", function() {
-        self.geometry = convertLineStringToPolygon(self.geometry);
-        self.redraw(self.geometry);
-        self.endDrawing(self.geometry, self.geoJson);
-      });
-    }
-  );
-});
+function clearButtonsState() {
+  document.getElementsByClassName("choice-button").forEach(button => {
```

Comment:
Bear in mind that arrow functions won't work in IE11.

Reply:
Yeah, I'm aware of this and we decided to not support IE in favor of having better readable code.